From e99dfd5b66cce2c8796ce9d1194761f97537fd4b Mon Sep 17 00:00:00 2001
From: Hieu Huynh <hieuh@os.amperecomputing.com>
Date: Fri, 19 Sep 2025 03:43:49 +0000
Subject: [PATCH] configurations: update Backplane's configurations

Jefferson 1P supports SKU1 with 12 x 2.5 NVMe. With current
configuration, the Backplane 2 shows 8 NVMe is not correct.
This commit updates Backplane 2 to show only 4 NVMe.

Tested: busctl tree xyz.openbmc_project.NVMeSensor
shows 12 NVMe sensors.

Upstream-Status: Pending

Signed-off-by: Hieu Huynh <hieuh@os.amperecomputing.com>
---
 configurations/mtjefferson_bp.json | 217 +++++++++++++++++++----------
 1 file changed, 140 insertions(+), 77 deletions(-)

diff --git a/configurations/mtjefferson_bp.json b/configurations/mtjefferson_bp.json
index 15343b0..2e5d9fe 100644
--- a/configurations/mtjefferson_bp.json
+++ b/configurations/mtjefferson_bp.json
@@ -1,81 +1,144 @@
-{
-    "Exposes": [
-        {
-            "Address": "$address",
-            "Bus": "$bus",
-            "Name": "Jefferson BP_EEPROM$ADDRESS / 82 + 1",
-            "Type": "EEPROM"
+[
+    {
+        "Exposes": [
+           {
+                "Address": "$ADDRESS",
+                "Bus": "$BUS",
+                "Name": "Mt.Jefferson BP 1 EEPROM",
+                "Type": "EEPROM"
+            },
+            {
+                "Address": "0x18",
+                "Bus": "0x9",
+                "EntityId": "0xf",
+                "EntityInstance": "0x2",
+                "Name": "FrontBPTemp1",
+                "Type": "JC42"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x30",
+                "Name": "nvme0",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x31",
+                "Name": "nvme1",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x32",
+                "Name": "nvme2",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x33",
+                "Name": "nvme3",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x34",
+                "Name": "nvme4",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x35",
+                "Name": "nvme5",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x36",
+                "Name": "nvme6",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x37",
+                "Name": "nvme7",
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "Mt.Jefferson BP 1",
+        "Probe": [
+            "xyz.openbmc_project.FruDevice({'BOARD_PART_NUMBER': 'CBP2980', 'ADDRESS': 80})",
+            "AND",
+            "FOUND('Mt.Jefferson_Motherboard')"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Decorator.Asset": {
+            "Manufacturer": "$BOARD_MANUFACTURER",
+            "Model": "$BOARD_PRODUCT_NAME",
+            "PartNumber": "$BOARD_PART_NUMBER",
+            "SerialNumber": "$BOARD_SERIAL_NUMBER"
         },
-        {
-            "Address": "$ADDRESS % 80 + 24",
-            "Bus": "$bus",
-            "Name": "FrontBPTemp$ADDRESS / 82 + 1",
-            "Type": "JC42"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 0",
-            "Name": "nvme$ADDRESS / 82 * 8 + 0",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 1",
-            "Name": "nvme$ADDRESS / 82 * 8 + 1",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 2",
-            "Name": "nvme$ADDRESS / 82 * 8 + 2",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 3",
-            "Name": "nvme$ADDRESS / 82 * 8 + 3",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 4",
-            "Name": "nvme$ADDRESS / 82 * 8 + 4",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 5",
-            "Name": "nvme$ADDRESS / 82 * 8 + 5",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 6",
-            "Name": "nvme$ADDRESS / 82 * 8 + 6",
-            "Type": "NVME1000"
-        },
-        {
-            "Address": "0x6a",
-            "Bus": "$ADDRESS / 82 * 8 + 48 + 7",
-            "Name": "nvme$ADDRESS / 82 * 8 + 7",
-            "Type": "NVME1000"
+        "xyz.openbmc_project.Inventory.Decorator.Replaceable": {
+            "FieldReplaceable": false,
+            "HotPluggable": false
         }
-    ],
-    "Name": "Jefferson BP_$ADDRESS / 82 + 1",
-    "Probe": [
-        "xyz.openbmc_project.FruDevice({'BOARD_PART_NUMBER': 'CBP2980'})",
-        "AND",
-        "FOUND('Mt.Jefferson_Motherboard')"
-    ],
-    "Type": "Board",
-    "xyz.openbmc_project.Inventory.Decorator.Asset": {
-        "Manufacturer": "$BOARD_MANUFACTURER",
-        "Model": "$BOARD_PRODUCT_NAME",
-        "PartNumber": "$BOARD_PART_NUMBER",
-        "SerialNumber": "$BOARD_SERIAL_NUMBER"
     },
-    "xyz.openbmc_project.Inventory.Decorator.Replaceable": {
-        "FieldReplaceable": false,
-        "HotPluggable": false
+    {
+        "Exposes": [
+            {
+                "Address": "$ADDRESS",
+                "Bus": "$BUS",
+                "Name": "Mt.Jefferson BP 2 EEPROM",
+                "Type": "EEPROM"
+            },
+            {
+                "Address": "0x1a",
+                "Bus": "0x9",
+                "EntityId": "0xf",
+                "EntityInstance": "0x3",
+                "Name": "FrontBPTemp2",
+                "Type": "JC42"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x38",
+                "Name": "nvme8",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x39",
+                "Name": "nvme9",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x3a",
+                "Name": "nvme10",
+                "Type": "NVME1000"
+            },
+            {
+                "Address": "0x6a",
+                "Bus": "0x3b",
+                "Name": "nvme11",
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "Mt.Jefferson BP 2",
+        "Probe": [
+            "xyz.openbmc_project.FruDevice({'BOARD_PART_NUMBER': 'CBP2980', 'ADDRESS': 82})",
+            "AND",
+            "FOUND('Mt.Jefferson_Motherboard')"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Decorator.Asset": {
+            "Manufacturer": "$BOARD_MANUFACTURER",
+            "Model": "$BOARD_PRODUCT_NAME",
+            "PartNumber": "$BOARD_PART_NUMBER",
+            "SerialNumber": "$BOARD_SERIAL_NUMBER"
+        },
+        "xyz.openbmc_project.Inventory.Decorator.Replaceable": {
+            "FieldReplaceable": false,
+            "HotPluggable": false
+        }
     }
-}
+]
-- 
2.43.0

